broker-daemon/block-order-worker/index.js

```diff
@@ -310,28 +310,33 @@ class BlockOrderWorker extends EventEmitter {
    * @returns {Object} contains activeOutboundAmount and activeInboundAmount of orders/fills
    */
   async calculateActiveFunds (marketName, side) {
-    const blockOrders = await this.getBlockOrders(marketName)
-    const blockOrdersForSide = blockOrders.filter(blockOrder => blockOrder.side === side)
-
-    const fullBlockOrders = await Promise.all(
-      blockOrdersForSide.map(async (bo) => {
-        await Promise.all([
-          bo.populateOrders(this.ordersStore),
-          bo.populateFills(this.fillsStore)
-        ])
-        return bo
-      })
-    )
-
-    let activeOutboundAmount = Big(0)
-    let activeInboundAmount = Big(0)
+    const blockOrders = await this.getBlockOrders(marketName, { side })
 
-    for (let blockOrder of fullBlockOrders) {
-      activeOutboundAmount = activeOutboundAmount.plus(blockOrder.activeOutboundAmount())
-      activeInboundAmount = activeInboundAmount.plus(blockOrder.activeInboundAmount())
+    // If we have no block orders that are available for our particular side, then
+    // we assume no active orders are available and can simply return 0
+    if (!blockOrders.length) {
+      return { activeOutboundAmount: Big(0), activeInboundAmount: Big(0) }
     }
 
-    return { activeOutboundAmount, activeInboundAmount }
+    // We grab the firstId and lastId to create a blockorder range so that we can
+    // grab all fills and orders to calculate active funds
+    const firstId = blockOrders[blockOrders.length - 1].id
```

Comment:
Are the block orders guaranteed to be sorted by ID?

Reply:
Block order IDs should be sorted by time so long as there weren't any orders that went through an order lifecycle prior to the lexicographical ordering update.

Reply:
I guess what I mean is: does `getBlockOrders` make an API guarantee that block orders will be streamed in order? Maybe it does, but we should document that at least if we are relying on it.

Reply:
No longer need to worry about this because I removed the call to blockorders